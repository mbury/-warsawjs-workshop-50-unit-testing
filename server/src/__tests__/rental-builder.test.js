const Rental = require('../rentals/rental');
const RentalBuilder = require('../rentals/rental-builder');

const builder = new RentalBuilder();

describe('Rental builder', () => {
  test('should return valid rental class', () => {
    const rental = builder
      .rentBy(1)
      .selectCar(1)
      .depositAmount(5000)
      .inState(Rental.RESERVED)
      .build();
    expect(rental).toMatchSnapshot();
  });

  test('should return class with valid payed state', () => {
    const rental = builder.buildPaid();
    expect(rental.getState()).toMatchInlineSnapshot(`"DEPOSIT_PAID"`);
  });
});
