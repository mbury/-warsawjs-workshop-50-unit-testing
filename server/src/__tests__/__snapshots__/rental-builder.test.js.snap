// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rental builder should return valid rental class 1`] = `
Rental {
  "car_id": 1,
  "client_id": 1,
  "deposit": 5000,
  "rental_id": 1,
  "state": "RESERVED",
}
`;
